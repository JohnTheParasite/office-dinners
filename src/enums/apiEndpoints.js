export class ApiEndpoints {
  static LOGIN = process.env.VUE_APP_API_URL + "/login"
  static USER_LIST = process.env.VUE_APP_API_URL + "/user/list"
  static CREATE_USER = process.env.VUE_APP_API_URL + "/user/create"
  static USER_DATA = process.env.VUE_APP_API_URL + "/user/update"
  static CAFE_LIST = process.env.VUE_APP_API_URL + "/cafe/list"
  static TOGGLE_CAFE = process.env.VUE_APP_API_URL + "/cafe/toggle"
  static CREATE_CAFE = process.env.VUE_APP_API_URL + "/cafe/create"
  static CAFE_DATA = process.env.VUE_APP_API_URL + "/cafe/update"
  static CAFE_COMMENTS = process.env.VUE_APP_API_URL + "/comment/cafe"
  static CREATE_COMMENT = process.env.VUE_APP_API_URL + "/comment/add"
  static VOTE = process.env.VUE_APP_API_URL + "/cafe/vote"
  static CAFE_HEADER = process.env.VUE_APP_API_URL + "/cafe/header-data"
  static VOTES_OPEN = process.env.VUE_APP_API_URL + "/cafe/open-votes"
  static VOTES_CLOSE = process.env.VUE_APP_API_URL + "/cafe/close-votes"
  static VOTES_AUTOCLOSE = process.env.VUE_APP_API_URL + "/cafe/set-auto-close-time"
  static ORDER_ADD = process.env.VUE_APP_API_URL + "/cafe/add-order"
  static ORDER_CHANGE = process.env.VUE_APP_API_URL + "/order/change-order"
  static ORDERS_GET = process.env.VUE_APP_API_URL + "/order/get-for-date"
  static SET_ORDER_USER_TO_PAY = process.env.VUE_APP_API_URL + "/order/select-user-to-pay"
  static CLOSE_ORDER = process.env.VUE_APP_API_URL + "/order/close-order"
  static OPEN_ORDER = process.env.VUE_APP_API_URL + "/order/open-order"
  static SET_ORDER_ORDERED = process.env.VUE_APP_API_URL + "/order/set-ordered"
  static DELETE_ORDER = process.env.VUE_APP_API_URL + "/order/delete"
  static UPDATE_ORDER_PRICES = process.env.VUE_APP_API_URL + "/order/update-order-prices"
  static SET_ORDER_DISCOUNT = process.env.VUE_APP_API_URL + "/order/set-order-discount"
}
