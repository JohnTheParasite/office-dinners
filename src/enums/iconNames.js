export default class IconNames {
  static EYE = "fa-eye"
  static EYE_SLASH = "fa-eye-slash"
}
