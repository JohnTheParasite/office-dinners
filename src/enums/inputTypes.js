export default class InputTypes {
  static TEXT = "text"
  static PASSWORD = "password"
}
