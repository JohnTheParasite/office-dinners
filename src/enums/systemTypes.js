export default class SystemTypes {
  static DEFAULT = "default"
  static SUCCESS = "success"
  static DANGER = "danger"
  static WARNING = "warning"
  static PRIMARY = "primary"
}
