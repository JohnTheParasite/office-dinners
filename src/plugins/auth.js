import AuthService from "@/services/authService";

export const Auth = {
  install(Vue) {
    Vue.prototype.$authService = AuthService;
  },
};