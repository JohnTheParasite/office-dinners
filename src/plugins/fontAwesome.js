import { library } from "@fortawesome/fontawesome-svg-core"
import {
  faArrowRightToBracket,
  faArrowUpRightFromSquare,
  faBan,
  faBurger,
  faChartLine,
  faCheck,
  faChevronDown,
  faChevronLeft,
  faChevronRight,
  faEye,
  faEyeSlash,
  faHandHoldingDollar,
  faHeart,
  faPencil,
  faPercent,
  faTrashCan,
  faUser,
  faUsers,
  faUtensils,
  faXmark
} from "@fortawesome/free-solid-svg-icons"
import { FontAwesomeIcon } from "@fortawesome/vue-fontawesome"
import Vue from "vue"

library.add(faChevronLeft)
library.add(faChevronRight)
library.add(faChevronDown)
library.add(faArrowUpRightFromSquare)
library.add(faCheck)
library.add(faPencil)
library.add(faTrashCan)
library.add(faBan)
library.add(faPercent)
library.add(faChartLine)
library.add(faUsers)
library.add(faBurger)
library.add(faHandHoldingDollar)
library.add(faUtensils)
library.add(faHeart)
library.add(faUser)
library.add(faArrowRightToBracket)
library.add(faXmark)
library.add(faEye)
library.add(faEyeSlash)
Vue.component("font-awesome-icon", FontAwesomeIcon)
