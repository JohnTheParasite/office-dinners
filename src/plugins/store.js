import store from "@/store/store"

export const Store = {
  install(Vue) {
    Vue.prototype.$store = store
  }
}
