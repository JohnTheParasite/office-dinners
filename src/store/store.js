import Vue from "vue"
import Vuex from "vuex"
import global from "./modules/global"
import toasts from "./modules/toasts"

Vue.use(Vuex)

export default new Vuex.Store({
  strict: true,
  state: {},
  mutations: {},
  actions: {},
  modules: {
    global,
    toasts
  }
})
