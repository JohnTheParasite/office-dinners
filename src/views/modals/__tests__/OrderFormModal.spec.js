import { createLocalVue, mount, shallowMount } from "@vue/test-utils"
import orderFormModal from "@/views/modals/OrderFormModal"
import i18n from "@/i18n"
import store from "@/store/store"
import { BootstrapVue } from "bootstrap-vue"
import AuthService from "@/services/authService"
import "@/plugins/fontAwesome"
import { ApiEndpoints } from "@/enums/apiEndpoints"

describe("orderFormModal", () => {
  AuthService.getUserData = () => {
    return { id: 2 }
  }

  const axios = require("axios")
  jest.mock("axios")

  const localVue = new createLocalVue()
  localVue.use(i18n)
  localVue.use(BootstrapVue)

  it("should render properly", () => {
    let orderForm = shallowMount(orderFormModal, { localVue, i18n, store, mocks: { $authService: AuthService } })
    expect(orderForm).toMatchSnapshot()
    orderForm.vm.formGroup.order_name = "test"
    orderForm.vm.formGroup.price = 15
    expect(orderForm.vm.verified).toBeTruthy()
  })

  it("should test show() method", async () => {
    let orderForm = mount(orderFormModal, { localVue, i18n, store, mocks: { $authService: AuthService } })
    expect(orderForm.vm.formGroup.name).toBe("")
    orderForm.vm.show({ name: "test" })
    await orderForm.vm.$nextTick()
    expect(orderForm.vm.$children[0].isVisible).toBeTruthy()
    expect(orderForm.vm.formGroup.name).toBe("test")
  })

  it("should test onChange() method", async () => {
    let orderForm = mount(orderFormModal, { localVue, i18n, store, mocks: { $authService: AuthService } })
    expect(orderForm.vm.formGroup.name).toBe("")
    orderForm.vm.onChange("name", "test")
    expect(orderForm.vm.formGroup.name).toBe("test")
  })

  it("should test add() method", async () => {
    let orderForm = mount(orderFormModal, {
      localVue,
      i18n,
      store,
      mocks: { $authService: AuthService, $axios: axios }
    })
    orderForm.vm.show({ name: "test" })
    await orderForm.vm.$nextTick()
    expect(orderForm.vm.$children[0].isVisible).toBeTruthy()

    axios.post.mockResolvedValueOnce({ data: true })
    await orderForm.vm.add()
    expect(axios.post).toHaveBeenCalledWith(ApiEndpoints.ORDER_ADD, expect.anything())
    await orderForm.vm.$nextTick()
    expect(orderForm.vm.$children[0].isVisible).toBeFalsy()

    axios.post.mockRejectedValueOnce(false)
    await orderForm.vm.add()
    expect(axios.post).toHaveBeenCalledWith(ApiEndpoints.ORDER_ADD, expect.anything())
  })
})
