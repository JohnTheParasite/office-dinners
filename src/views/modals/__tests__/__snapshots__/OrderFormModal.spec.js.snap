// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`orderFormModal should render properly 1`] = `
<b-modal-stub id="orderDataModal" canceltitle="Cancel" cancelvariant="secondary" centered="true" headerclass="primary" headerclosecontent="&amp;times;" headercloselabel="Close" nocloseonbackdrop="true" oktitle="OK" okvariant="primary" size="lg" titletag="h5">
  <div class="cafe-link-rating"><a href="" target="_blank">order.clicklink
      <font-awesome-icon-stub icon="fa-solid fa-arrow-up-right-from-square"></font-awesome-icon-stub>
    </a>
    <b-form-rating-stub value="0" iconclear="x" iconempty="star" iconfull="star-fill" iconhalf="star-half" readonly="true" stars="5"></b-form-rating-stub>
  </div>
  <div class="date-delivery-rating">
    <div class="date">
      <div class="last-order-label">table.columns.last_order_date:</div>
      <div class="last-order-date"></div>
    </div>
    <b-form-rating-stub value="0" iconclear="x" iconempty="star" iconfull="star-fill" iconhalf="star-half" readonly="true" stars="5"></b-form-rating-stub>
  </div>
  <div class="inputs">
    <text-input-stub label="order.order" required="true" description="" name="" initvalue="" type="text" placeholder="" warningmessage="errors.fieldIsRequired" icon="" oniconclick="[Function]" class="order"></text-input-stub>
    <text-input-stub label="order.price" required="true" description="" name="" initvalue="" type="Number" placeholder="" warningmessage="errors.fieldIsRequired" icon="" oniconclick="[Function]"></text-input-stub>
  </div>
  <!---->
</b-modal-stub>
`;
